src/jobs/java_maven_build_test.yml

```diff
@@ -0,0 +1,48 @@
+description: >
+  Build and test java app using maven; update circleCI caches and workspace.
+  Supports app-module from monorepo as well as single app from single-app-repo.
+
+parameters:
+  app_path:
+    description: "Path of maven module for the app, or \".\" for single-app-repo"
+    type: string
+    default: "."
+  maven_cache_key_prefix:
+    description: "Prefix for the maven artifacts cache-key (used in combination with checksum of pom.xml); no caching with blank prefix"
+    type: string
+    default: ""
+  executor:
+    description: "Executor for the build (java_builder_docker is more lightweight, but some tests leveraging testcontainers etc require the java_builder_vm)"
+    type: enum
+    enum: [ "java_builder_docker", "java_builder_vm"]
+    default: "java_builder_docker"
+  java_builder_image:
+    description: "Image to use for the executor"
+    type: string
+    default: "cimg/openjdk:11.0"
```

Comment:
that parameter makes the executor configurable in the java-application config

if i want to choose the executor by parameter and still be able to overwrite executor parameters (image version), the extension points of circleCI really dont provide what is needed. i cant even add a conditional when selecting/parametrizing the executor!

so what i did was
- select the executor by name (`java_builder_*` parameter)
- have a common set of params for all `java_builder_*` exectuors so that i can inject the required parameters/defaults without knowing where the values actually go. - a.k.a. polymorphism

if we dont need the power to overwrite the images in the java-repository configs, we can simplify here. but as long as we want to keep that capability there is no other way to do this AFAIK

Reply:
simplified now - by hardcoding the executor parametrization we can simplify a lot here